mavlink_vehicles/mavlink_vehicles.cc

```diff
@@ -1167,6 +1173,7 @@ void mav_vehicle::update()
 
         // Get back to AUTO mode to continue the mission
         if (this->detour_waypoint_autocontinue) {
+            this->is_our_control = true;
```

Comment:
Does it have any conflict if you use 'git rebase -e' to move this line to "Detect when someone else has taken control"